hierarchicalforecast/utils.py

```diff
@@ -258,27 +260,40 @@ def _to_summing_dataframe(df: pd.DataFrame,
             bottom_col = '/'.join(hier)
             df['unique_id'] = df[hier].agg('/'.join, axis=1)
         else:
-            hier_col = '/'.join(hier) 
+            hier_col = '/'.join(hier)
         S_df[hier_col] = S_df[hier].agg('/'.join, axis=1)
         hiers_cols.append(hier_col)
     S_df = S_df.sort_values(by=bottom_comb)
     S_df = S_df[hiers_cols]
 
     #------------------------------- Encoding ------------------------------#
     # One hot encode only aggregate levels
-    # TODO: option to only operate with sparse matrices
     bottom_ids = list(S_df.unique_id)
     del S_df['unique_id']
     categories = [S_df[col].unique() for col in S_df.columns]
     tags = dict(zip(S_df.columns, categories))
     tags[bottom_col] = bottom_ids
 
-    encoder = OneHotEncoder(categories=categories,
-                            sparse=False, dtype=np.float32)
+    encoder = OneHotEncoder(
```

Comment:
The only change in OneHotEncoder is that they changed the argument name from `sparse` to `sparse_output` (likely due to the conflict with scipy.sparse package; I actually got a crazy nasty error because of this name conflict once). I am avoiding creating S as dense matrix as much as possible, as for really large datasets (which I'm ultimately aiming at) it will oom immediately.